crates/worker/src/partition_processor_manager.rs

```diff
@@ -770,83 +778,83 @@ impl PartitionProcessorManager {
         partition_table: &PartitionTable,
     ) {
         let partition_id = control_processor.partition_id;
-
-        match control_processor.command {
-            ProcessorCommand::Stop => {
-                if let Some(processor_state) = self.processor_states.get_mut(&partition_id) {
-                    debug!(%partition_id, "Asked to stop partition processor by cluster controller");
-                    processor_state.stop();
-                }
-                if self.pending_snapshots.contains_key(&partition_id) {
-                    info!(%partition_id, "Partition processor stop requested with snapshot task result outstanding");
-                }
-                self.archived_lsns.remove(&partition_id);
-                self.latest_snapshots.remove(&partition_id);
-            }
-            ProcessorCommand::Follower | ProcessorCommand::Leader => {
-                if let Some(processor_state) = self.processor_states.get_mut(&partition_id) {
-                    if control_processor.command == ProcessorCommand::Leader {
-                        if let Some(leader_epoch_token) = processor_state.run_as_leader() {
-                            debug!(%partition_id, "Asked to run as leader by cluster controller. Obtaining required leader epoch");
-                            Self::obtain_new_leader_epoch(
-                                partition_id,
-                                leader_epoch_token,
-                                self.metadata_writer.raw_metadata_store_client().clone(),
-                                &mut self.asynchronous_operations,
-                            );
-                        }
-                    } else if control_processor.command == ProcessorCommand::Follower {
-                        debug!(%partition_id, "Asked to run as follower by cluster controller");
-                        if let Err(err) = processor_state.run_as_follower() {
-                            info!(%partition_id, %err, "Partition processor failed to run as follower. Stopping it now");
-                            processor_state.stop();
-                        }
+        info_span!("on_control_processor", %partition_id).in_scope(|| {
```

Comment:
is that different from instrumenting the function itself `#[instrument(..)]` and extracting the partition id from `control_processor`?

Reply:
No it is not. Somehow I was thinking that `#[instrument]` only works on async functions. What a brain fart. Will correct.